samples/algorithms/sudoku-grover/Sudoku.qs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.Quantum.Samples.SudokuGrover {
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Samples.ColoringGroverWithConstraints;
+
+    /// # Summary
+    /// Solve a Sudoku puzzle using Grover's algorithm
+    ///
+    ///
+    /// # Description
+    /// Sudoku is a graph coloring problem where graph edges must connect nodes 
+    /// of different colors
+    /// In our case, Graph Nodes are puzzle squares and colors are the Sudoku numbers. 
+    /// Graph Edges are the constraints preventing squares from having the same values. 
+    /// To reduce the number of qubits needed, we only use qubits for empty squares
+    /// We define the puzzle using 2 data structures.
+    ///   - A list of edges connecting empty squares
+    ///   - A list of constraints on empty squares to the initial numbers 
+    ///     in the puzzle (starting numbers)
```

Comment:
Markdown bulleted lists should be surrounded by empty lines to avoid them getting merged with paragraphs above and below.
```suggestion
    ///
    ///   - A list of edges connecting empty squares
    ///   - A list of constraints on empty squares to the initial numbers 
    ///     in the puzzle (starting numbers)
    ///
```